### PR TITLE
LAU-484 add autoselect field for jurisdictions and case types

### PR DESCRIPTION
**Pull request by rapolaskaseliscgi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/LAU-484

### Change description ###
Replaced input field with autoselect field for jurisdictions and case types

### CODE REVIEW

**Review by MartinYSpasov — APPROVED** [review]

File: src/main/app.ts [hunk 1]
```diff
@@ -40,12 +41,15 @@ new CSRFToken().enableFor(app);
...
 new HealthCheck().enableFor(app);
+new Container().enableFor(app);
 
 logger.info('Environment: ' + env);
 
-setupDev(app,developmentMode);
+setupDev(app, developmentMode);
 setupTest(app);
 
+app.locals.container.cradle.autoSuggestService.loadData(developmentMode || env === 'test');
```
Comment (sonny-hmcts) [inline src/main/app.ts:51]: I would look into moving the loading of autoSuggestService into its own module. this module could then do something like the following

Module
```
export class AutoSuggest {

  constructor(
   private autoSuggestService: autoSuggestInterface
   private developmentMode: boolean
   ) {}
  
 enableFor(app: express.Express): void {

    app.use((req, res, next) => {
    //res.locals are available throughout the application including nunjucks so no need to pass it through on all methods
     res.locals.jurisdictions = autoSuggestService.getJurisdictions();
     res.locals.caseTypes = autoSuggestService.getCaseTypes();
    })
  }

}
```

then in application.ts

```
new AutoSuggest(app.locals.container.cradle.autoSuggestService, developmentMode || env == 'test').enableFor(app);
```

File: src/main/routes/home.ts [hunk 2]
```diff
@@ -77,6 +80,8 @@ async function caseHandler(req: AppRequest, res: Response) {
     caseSearchPage: true,
     caseActivities: req.session?.caseActivities,
     caseSearches: req.session?.caseSearches,
+    jurisdictions: app.locals.container.cradle.autoSuggestService.getJurisdictionsData(),
+    caseTypes: app.locals.container.cradle.autoSuggestService.getCaseTypesData(),
```
Comment (sonny-hmcts) [inline src/main/routes/home.ts:84]: see the above comment about putting the loading of data into a module. 

the example given on setting the data

```res.locals.CaseType = data```

is available throughout the application and nunjucks this means that there is no need to do the following

```
jurisdictions: app.locals.container.cradle.autoSuggestService.getJurisdictionsData(),
caseTypes: app.locals.container.cradle.autoSuggestService.getCaseTypesData(),
```

File: src/main/views/case-audit/template.njk [hunk 3]
```diff
@@ -51,25 +51,28 @@
...
         label: {
           text: "Jurisdiction ID"
         },
-        classes: "govuk-!-width-three-quarters",
-        id: "caseJurisdictionId",
-        name: "caseJurisdictionId",
-        value: caseForm.caseJurisdictionId
+        classes: "govuk-!-width-three-quarters accessible-autocomplete",
+        value: caseForm.caseJurisdictionId,
+        items: jurisdictions
```
Comment (sonny-hmcts) [inline src/main/views/case-audit/template.njk:73]: as mentioned above the 

``` res.locals.jurisdictions = data ```

is available directly in nunjucks. the above will be accessible via

``` jurisdictions ```

**Review by sonny-hmcts — COMMENTED** [review]
please see notes.

**Review by sonny-hmcts — APPROVED** [review]

**Pull request merged**